python/tskit/genotypes.py

```diff
@@ -26,16 +26,23 @@
 import numpy as np
 
 import _tskit
+import tskit
 import tskit.trees as trees
 
 
 class Variant:
     """
     A variant in a tree sequence, describing the observed genetic variation
-    among samples for a given site. A variant consists (a) of a reference to
-    the :class:`Site` instance in question; (b) the **alleles** that may be
-    observed at the samples for this site; and (c) the **genotypes**
-    mapping sample IDs to the observed alleles.
+    among samples for a given site. A variant consists of (a) the **alleles** that
+    may be observed at the sample nodes for this site; (b) the **genotypes**
+    mapping sample IDs to the observed alleles and (c) of a reference to
+    the :class:`Site` that the Variant has been decoded at.
+
+    After creation a Variant is not yet decoded, and has no genotypes.
+    To decode a Variant, call the :meth:`decode` method. The Variant class will then
+    use a Tree, internal to the Variant, to seek to the position of the site and decode
+    the genotypes at that site. As such it is currently much more efficient to decode
```

Comment:
I think it'll always be more efficient to do it this way, indexes will just make it less inefficient to randomly seek. I'd say "It is therefore much more efficient to sites in sequential genomic order, either in a forwards or backwards direction, than to do so randomly".